src/data_utils/other.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Returns a name that is safe for stuff such as links.
+ * 
+ * 1. Remove extra whitespace around the name
+ * 2. Replace all spaces with underscores ("`_`")
+ * 3. Converts to lowercase.
+ * 
+ * @param name The name to convert.
+ * @returns The safe name.
+ */
+export function toSafeName(name: string) {
+    return name.trim().replaceAll(" ", "_").toLowerCase();
+}
+
+
+/**
+ * Returns `true` if the string is not `undefined` and not `""`, otherwise `false`.
+ * 
+ * @param value The value to check.
+ * @returns The result of the check.
+ */
+export function isBlankString(value: string | undefined): boolean {
+    return value === undefined || value === ""
+}
+
+/**
+ * Returns the trimmed string if the value is not `undefined` or an empty string, otherwise returns `undefined`.
+ * 
+ * @param value The value to check.
+ * @returns The trimmed {@link value} if {@link value} is not empty, otherwise `undefined`.
+ */
+export function ensureNotBlankString(value: string | undefined): string | undefined {
+    return (value !== undefined && !isBlankString(value)) ? value.trim() : undefined;
```

Comment:
Can we not use `isBlankString()` and inverse the result?

Reply:
some of the `ensure` functions have an extra `value !== undefined` tacked onto them since typescript somewhat doesn't infer that `value` can't ever be `undefined` after the initial check in the function